ReactiveCocoaTests/Swift/SignalProducerSpec.swift

```diff
@@ -373,6 +373,55 @@ class SignalProducerSpec: QuickSpec {
 				expect(values).to(equal([ 1, 2, 3 ]))
 			}
 
+			it("should not deadlock in pair when started while sending") {
+				let (producer1, observer1) = SignalProducer<String, NoError>.buffer()
+				let (producer2, observer2) = SignalProducer<String, NoError>.buffer()
+
+				observer1.sendNext("A")
+				observer1.sendNext("B")
+				observer2.sendNext("1")
+				observer2.sendNext("2")
+
+				var valuePairs: [String] = []
+				producer1.startWithCompleted {
+					producer2.startWithCompleted {
+						valuePairs = []
+						producer1.startWithNext { value1 in
+							producer2.startWithNext { value2 in
+								valuePairs.append(value1 + value2)
+							}
+						}
+					}
+				}
+
+				observer1.sendCompleted()
+				observer2.sendCompleted()
+				expect(valuePairs).to(equal([ "A1", "A2", "B1", "B2" ]))
+			}
+
+			it("should send values in the order they were buffered") {
+				let (producer, observer) = SignalProducer<Int, NoError>.buffer()
+
+				observer.sendNext(1)
+				observer.sendNext(2)
+
+				var values: [Int] = []
+				producer.startWithNext { value in
+					values.append(value)
+					if value == 1 {
+						// the expected behavior is for the other queue to block until
```

Comment:
I'm not sure if this is the right way to test for concurrency effects or even if RAC supports this kind of concurrent access (I guess that was the idea behind all those locks, but I never considering such usage in my projects).